test/unit/app/controllers/permissions/permissions-controller-test.js

```diff
@@ -0,0 +1,1208 @@
+import { strict as assert } from 'assert'
+import { find } from 'lodash'
+
+import {
+  METADATA_STORE_KEY,
+  WALLET_PREFIX,
+} from '../../../../../app/scripts/controllers/permissions/enums'
+
+import getRestrictedMethods
+  from '../../../../../app/scripts/controllers/permissions/restrictedMethods'
+
+import {
+  addInternalMethodPrefix,
+} from '../../../../../app/scripts/controllers/permissions'
+
+import {
+  constants,
+  getters,
+  getApprovedPermissionsRequest,
+  getNotifyDomain,
+  getNotifyAllDomains,
+  getPermController,
+  grantPermissions,
+} from './mocks'
+
+const {
+  ERRORS,
+  NOTIFICATIONS,
+  PERMS,
+} = getters
+
+const {
+  ACCOUNT_ARRAYS,
+  DUMMY_ACCOUNT,
+  ORIGINS,
+  PERM_NAMES,
+  REQUEST_IDS,
+} = constants
+
+const initNotifications = () => {
+  return Object.values(ORIGINS).reduce((acc, domain) => {
+    acc[domain] = []
+    return acc
+  }, {})
+}
+
+const initPermController = (notifications = initNotifications()) => {
+  return getPermController({
+    notifyDomain: getNotifyDomain(notifications),
+    notifyAllDomains: getNotifyAllDomains(notifications),
+  })
+}
+
+const getMockRequestUserApprovalFunction = (permController) => (id) => {
+  return new Promise((resolve, reject) => {
+    permController.pendingApprovals[id] = { resolve, reject }
+  })
+}
+
+describe('permissions controller', function () {
+
+  describe('getAccounts', function () {
+
+    let permController
+
+    beforeEach(function () {
+      permController = initPermController()
+      grantPermissions(
```

Comment:
As discussed, added a docstring for that utility method instead.

`permController.permissions.grantPermissions` exposes implementation details that are of no relevance to what's being tested.

Reply:
Those implementation details are still included in the test, just indirectly now rather than directly. Admittedly this will make the tests easier to update in the future if this changes, but I don't think it makes the tests easier to read or understand, or isolates them from implementation details. To fix that would require decoupling `rpc-cap` itself from this module.

I don't feel like pushing back on this any more as it doesn't really matter that much, but I still don't understand the utility of this method or the others like it 🤷‍♂